modules/gltf/editor/editor_import_blend_runner.cpp

```diff
@@ -0,0 +1,303 @@
+
+	// Connect to RPC server.
+	Ref<HTTPClient> client = HTTPClient::create();
+	client->connect_to_host("127.0.0.1", rpc_port);
+
+	bool done = false;
+	while (!done) {
+		HTTPClient::Status status = client->get_status();
+		switch (status) {
+			case HTTPClient::STATUS_RESOLVING:
+			case HTTPClient::STATUS_CONNECTING: {
+				client->poll();
+				break;
+			}
+			case HTTPClient::STATUS_CONNECTED: {
+				done = true;
+				break;
+			}
+			default: {
+				ERR_FAIL_V_MSG(ERR_CONNECTION_ERROR, vformat("Unexpected status during RPC connection: %d", status));
+			}
+		}
+	}
+
+	// Send XML request.
+	PackedByteArray xml_buffer = xml_body.to_utf8_buffer();
+	Error err = client->request(HTTPClient::METHOD_POST, "/", Vector<String>(), xml_buffer.ptr(), xml_buffer.size());
+	if (err != OK) {
+		ERR_FAIL_V_MSG(err, vformat("Unable to send RPC request: %d", err));
+	}
+
+	// Wait for response.
+	done = false;
+	while (!done) {
+		HTTPClient::Status status = client->get_status();
+		switch (status) {
+			case HTTPClient::STATUS_REQUESTING: {
+				client->poll();
+				break;
+			}
+			case HTTPClient::STATUS_BODY: {
+				client->poll();
+				// Parse response here if needed. For now we can just ignore it.
+				done = true;
+				break;
+			}
+			default: {
+				ERR_FAIL_V_MSG(ERR_CONNECTION_ERROR, vformat("Unexpected status during RPC response: %d", status));
+			}
+		}
+	}
+
+	return OK;
+}
+
+Error EditorImportBlendRunner::do_import_direct(const Dictionary &p_options) {
+	// Export glTF directly.
+	String python = vformat("import bpy, sys\n\
+opts = %s\n\
+if bpy.app.version < (3, 0, 0):\n\
+  print('Blender 3.0 or higher is required.', file=sys.stderr)\n\
+bpy.ops.wm.open_mainfile(filepath=opts['path'])\n\
+if opts['unpack_all']:\n\
+  bpy.ops.file.unpack_all(method='USE_LOCAL')\n\
+bpy.ops.export_scene.gltf(**opts['gltf_options'])",
+			dict_to_python(p_options));
+	Error err = start_blender(python, true);
+	if (err != OK) {
+		return err;
+	}
+
+	return OK;
+}
+
+void EditorImportBlendRunner::_resources_reimported(const PackedStringArray &p_files) {
+	if (is_running()) {
+		// After a batch of imports is done, wait a few seconds before trying to kill blender,
+		// in case of having multiple imports trigger in quick succession.
+		kill_timer->start();
+	}
+}
+
+void EditorImportBlendRunner::_kill_blender() {
+	kill_timer->stop();
+	if (is_running()) {
+		OS::get_singleton()->kill(blender_pid);
+	}
+	blender_pid = 0;
+}
+
+EditorImportBlendRunner *EditorImportBlendRunner::singleton = nullptr;
+
+EditorImportBlendRunner *EditorImportBlendRunner::get_singleton() {
+	return singleton;
+}
+
+EditorImportBlendRunner::EditorImportBlendRunner() {
+	singleton = this;
+
+	rpc_port = EDITOR_GET("filesystem/import/blender/rpc_port");
+
+	kill_timer = memnew(Timer);
+	EditorNode::get_singleton()->add_child(kill_timer);
```

Comment:
Should I turn `EditorImportBlendRunner` into a singleton `Node` and add the timer as a child of it? Or is it fine to keep this?

Reply:
I don't know if it's fine to instance EditorImportBlendRunner  a 1000 times.

Reply:
I want `EditorImportBlendRunner` to be a singleton either way, so you shouldn't want to instance it. But I guess it helps making the class more self-contained if it's a `Node` and doesn't randomly put stuff into `EditorNode`. :)